include/deal.II/base/aligned_vector.h

```diff
@@ -310,6 +342,81 @@ namespace internal
    * @relates AlignedVector
    */
   template <typename T>
+  class AlignedVectorCopy : private parallel::ParallelForInteger
+  {
+    static const std::size_t minimum_parallel_grain_size = 160000/sizeof(T)+1;
+  public:
+    /**
+     * Constructor. Issues a parallel call if there are sufficiently many
+     * elements, otherwise works in serial. Copies the data from the half-open
+     * interval between @p source_begin and @p source_end to array starting at
+     * @p destination (by calling the copy constructor with placement new). If
+     * the flag copy_source is set to @p true, the elements from the source
+     * array are simply copied. If it is set to @p false, the data is moved
+     * between the two arrays by invoking the destructor on the source range
+     * (preparing for a subsequent call to free).
+     */
+    AlignedVectorCopy (T *source_begin,
+                       T *source_end,
+                       T *destination,
+                       const bool copy_source)
```

Comment:
I guess we can remove the `copy_source` argument now that we have a dedicate copy function. Similarly in the `AlignedVectorMove` class.